src/stylesheets/components/_header.scss

```diff
@@ -0,0 +1,631 @@
+$z-index-nav:     9000;
+
+
+// Header ------------- //
+
+.usa-header {
+  width: 100%;
+  z-index: $z-index-header;
+
+  @include media($nav-width) {
+    border-bottom: 1px solid $color-gray-light;
+  }
+
+  a {
+    border-bottom: none;
+  }
+
+  em {
+    display: block;
+    font-size: $h5-font-size;
+    font-style: normal;
+    font-weight: $font-bold;
+    margin: 0;
+
+    @include media($nav-width) {
+      display: block;
+      font-size: 2.4rem;
+      line-height: $heading-line-height;
+    }
+  }
+
+  .usa-search {
+    margin-bottom: 1.5rem;
+
+    @include media($nav-width) {
+      float: right;
+      margin-bottom: 0;
+      max-width: 21.5rem;
+    }
+  }
+}
+
+.usa-navbar {
+  border-bottom: 1px solid $color-gray-light;
+  height: 4rem;
+
+  @include media($nav-width) {
+    border-bottom: none;
+    display: inline-block;
+    height: 10.3rem;
+  }
+}
+
+.usa-logo {
+  float: left;
+  line-height: 4rem;
+  margin-left: 1.5rem;
+
+  @include media($nav-width) {
+    line-height: 7.5rem;
+    margin-left: 0;
+    margin-top: 3.3rem;
+  }
+
+  a {
+    color: $color-base;
+    text-decoration: none;
+  }
+
+  br {
+    display: none;
+
+    @include media($nav-width) {
+      display: block;
+    }
+  }
+}
+
+// scss-lint:disable PropertyCount
+.usa-menu-btn {
+  @include button-unstyled;
+  background-color: $color-primary;
+  color: $color-white;
+  display: inline;
+  float: right;
+  font-size: $h6-font-size;
+  height: 4rem;
+  line-height: 4rem;
+  text-align: center;
+  text-decoration: none;
+  text-transform: uppercase;
+  width: 5.8rem;
+
+  @include media($nav-width) {
+    display: none;
+  }
+
+  &:hover {
+    color: $color-white;
+    background-color: $color-primary-darker;
+  }
+
+  &:visited {
+    color: $color-white;
+  }
+}
+// scss-lint:enable PropertyCount
+
+.usa-overlay {
+  @include position(fixed, 0);
+  background: $color-black;
+  opacity: 0;
+  transition: all 0.2s ease-in-out;
+  visibility: hidden;
+  z-index: $z-index-overlay;
+
+  &.is-visible {
+    opacity: 0.1;
+    visibility: visible;
+  }
+}
+
+// Header navigation ------------- //
+
+@mixin nav-border {
+  border-bottom: 0.7rem solid $color-primary;
+  padding-bottom: 1rem;
+}
+
+.usa-nav-link {
+  -moz-osx-font-smoothing: grayscale; // Antialiased font smoothing for light text on dark background
+  -webkit-font-smoothing: antialiased;
+
+  &:hover {
+    span {
+      @include media($nav-width) {
+        @include nav-border;
+      }
+    }
+  }
+}
+
+.usa-nav-container {
+  @include media($nav-width) {
+    @include outer-container();
+    @include padding(null $site-margins);
+    max-width: $site-max-width;
+  }
+}
+
+.usa-nav {
+  $sliding-panel-width: 26rem;
+  @include position(fixed, 0 0 0 auto);
+  @include size($sliding-panel-width 100%);
+  @include transform(translateX($sliding-panel-width));
+  background: $color-white;
+  border-left: 1px solid $color-gray-light;
+  border-right: 0;
+  display: flex;
```

Comment:
If flexbox doesn't work, it reverts to non-flexbox view which is just a different order of the search box to nav links on mobile. Since it functions, I think this is an OK degradation that's in line with progressive enhancement.